assets/src/edit-story/components/canvas/canvasElementDropzone.js

```diff
@@ -49,7 +49,18 @@ function CanvasElementDropzone({ children }) {
 
   const onDropHandler = useCallback(
     (e) => {
-      if (isDragType(e, 'resource/media') && !activeDropTargetId) {
+      // Handles onDrop for shapes.
+      if (isDragType(e, 'shape') && !activeDropTargetId) {
+        const props = JSON.parse(e.dataTransfer.getData('shape'));
+        const { x, y } = pageContainer?.getBoundingClientRect();
+        insertElement('shape', {
+          ...props,
+          x: editorToDataX(e.clientX - x - props.width / 2),
+          y: editorToDataY(e.clientY - y - props.height / 2),
```

Comment:
Note how images are placed exactly where they're dropped - shapes in this PR however change both size, position and color(?) when dropped compared to the preview shape shown.

Reply:
So it's easy to fix this so the color is the same (the color is white now when dropped). However, I am not sure how to deal with the size and position. The icons for each shape are a bit small so when you drop them it'll look tiny if you took that shape. Maybe if the drag and drop ghost image is being fixed in another issue, maybe this should be fixed in that issue as well?

Reply:
Just create a new separate issue for this and mark it with **_UX needed_**.

Reply:
#2237 Done